python/ray/worker.py

```diff
@@ -783,17 +783,21 @@ def _get_arguments_for_execution(self, function_name, serialized_args):
                 created one of the arguments failed.
         """
         arguments = []
+        object_ids = []
         for (i, arg) in enumerate(serialized_args):
             if isinstance(arg, ObjectID):
-                # get the object from the local object store
-                argument = self.get_object([arg])[0]
-                if isinstance(argument, RayError):
-                    raise argument
+                object_ids.append(arg)
             else:
                 # pass the argument by value
-                argument = arg
+                arguments.append(arg)
 
-            arguments.append(argument)
+        # get the objects from the local object store
+        values = self.get_object(object_ids)
+        for i, value in enumerate(values):
+            if isinstance(value, RayTaskError):
+                raise RayGetError(object_ids[i], value)
+
+        arguments.extend(values)
```

Comment:
This looks like it could leave `arguments` out-of-order if the list is a mix of object IDs and values, right? For instance, if `serialized_args = [1, object_id]`, we will end up with `arguments = [value_at_object_id, 1]`.